src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -81,13 +78,20 @@ internal sealed class EvaluationContext : EvaluationContextBase
                 previous.Compilation :
                 metadataBlocks.ToCompilation();
```

Comment:
I agree, ideally we'd have a single CreateTypeContext without 'previous'. For now though, this allows testing the re-use and it matches the pair of methods for CreateMethodContext. Let's consider splitting this out later.